packages/govuk-frontend/src/govuk/components/skip-link/skip-link.mjs

```diff
@@ -31,28 +32,43 @@ export class SkipLink extends GOVUKFrontendComponent {
     this.$module = $module
 
     // Check for linked element
-    const $linkedElement = this.getLinkedElement()
-    if (!$linkedElement) {
-      return
+    try {
+      const $linkedElement = this.getLinkedElement()
+      this.$linkedElement = $linkedElement
+    } catch (errorCause) {
+      const cause =
+        errorCause instanceof Error ? errorCause : new Error(String(errorCause))
+      throw new MissingElementError('The linked HTML element does not exist', {
+        cause
+      })
     }
 
-    this.$linkedElement = $linkedElement
     this.$module.addEventListener('click', () => this.focusLinkedElement())
   }
 
   /**
    * Get linked element
    *
    * @private
-   * @returns {HTMLElement | null} $linkedElement - DOM element linked to from the skip link
+   * @throws {Error} If the "href" attribute does not contain a hash
+   * @throws {Error} If the element with the specified ID does not exist
+   * @returns {HTMLElement} $linkedElement - DOM element linked to from the skip link
    */
   getLinkedElement() {
     const linkedElementId = this.getFragmentFromUrl()
     if (!linkedElementId) {
-      return null
+      throw new Error(
+        `The Skip Link's "href" attribute does not contain a hash`
```

Comment:
Should we match "The link" to ["The linked" wording above](https://github.com/alphagov/govuk-frontend/pull/4177/files#diff-fb3a25d42cbb3621d375cc34dc960447e84fa2a8729fd16e97e4c34edd9b17aeR41)

Or should we always add a component name?

```suggestion
        `The link "href" attribute does not contain a hash`
```

```suggestion
        `Skip link: $module "href" attribute does not contain a hash`
```

Reply:
A bit of bike shedding, since we'll do a content review of these, but I think the second option is more immediately understandable